plotsearch/models/plot_search.py

```diff
@@ -515,5 +534,31 @@ def answer_truncate(self):
         return self.answer if len(self.answer) < 50 else (self.answer[:50] + "..")
 
 
+class RelatedPlotApplication(TimeStampedSafeDeleteModel):
+    """
+    In Finnish: Liittyvät (tontti/alue)hakemukset
+    """
+
+    lease = models.ForeignKey(
+        "leasing.Lease",
+        related_name="related_plot_applications",
+        on_delete=models.CASCADE,
+    )
+    content_type = models.ForeignKey(
+        ContentType,
+        on_delete=models.CASCADE,
+        # Explicitly limits choice to these models, but can be extended to any model
+        limit_choices_to={"model__in": ["targetstatus", "areasearch"]},
+    )
+    object_id = models.PositiveBigIntegerField()
```

Comment:
Now after adding also plotsearch to the mix, it would be three columns for different models.
The downside for generic relation is the limit_choices_to part which requires a migration if changed. But at this stage I don't think there will be more models added.